src/Stratis.SmartContracts.CLR/EcRecoverProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using NBitcoin;
+using NBitcoin.Crypto;
+using Stratis.SmartContracts.Core.Hashing;
+
+namespace Stratis.SmartContracts.CLR
+{
+    /// <summary>
+    /// Holds logic for the equivalent of the ECRECOVER opcode.
+    ///
+    /// This is static for now but when we know more about how we are going to use it we will adjust as necessary.
+    /// </summary>
+    public class EcRecoverProvider : IEcRecoverProvider
+    {
+        private readonly Network network;
+
+        public EcRecoverProvider(Network network)
+        {
+            this.network = network;
+        }
+
+        private static uint256 GetUint256FromMessage(byte[] message)
+        {
+            return new uint256(HashHelper.Keccak256(message));
+        }
+
+        /// <summary>
+        /// Retrieves the base58 address of the signer of an ECDSA signature.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="signature">The ECDSA signature prepended with header information specifying the correct value of recId.</param>
+        /// <returns>The base58 address for the signer of a signature.</returns>
+        public Address GetSigner(byte[] message, byte[] signature)
+        {
+            // TODO: Error handling for incorrect signature format etc.
```

Comment:
I think this needs to be resolved before this can be used in production.

Reply:
@rowandh PubKey.RecoverCompact already detects incorrect format and raise an exception. 

There is something else to do in here ?